horovod/mxnet/__init__.py

```diff
@@ -68,6 +69,22 @@ def set_wd_mult(self, args_wd_mult):
         self._optimizer.set_wd_mult(args_wd_mult)
 
 
+class DistributedTrainer(mx.gluon.Trainer):
```

Comment:
Since this implements the Trainer API, are all APIs like save_state/load_state validated and tested?

Reply:
I manually tested save_states and load_states and it worked. The DistributedTrainer is a subclass of Trainer which we only overrided one function `_allreduce_grads`.  This function has been covered by the MNIST example which we run in Horovod CI. For other APIs, the Python inheritance rules should govern and are covered by our MXNet unit tests.